test/common/schemaform/FileField.unit.spec.jsx

```diff
@@ -0,0 +1,243 @@
+import React from 'react';
+import { expect } from 'chai';
+import SkinDeep from 'skin-deep';
+import sinon from 'sinon';
+import ReactTestUtils from 'react-dom/test-utils';
+
+import { DefinitionTester, getFormDOM } from '../../util/schemaform-utils.jsx';
+
+import FileField from '../../../src/js/common/schemaform/FileField';
+import fileUploadUI, { fileSchema } from '../../../src/js/common/schemaform/definitions/file';
+
+const formContext = {
+  setTouched: sinon.spy()
+};
+const requiredSchema = {};
+
+describe('Schemaform <FileField>', () => {
+  it('should render', () => {
+    const idSchema = {
+      $id: 'field'
+    };
+    const schema = {};
+    const uiSchema = fileUploadUI('Files');
+    const tree = SkinDeep.shallowRender(
+      <FileField
+          schema={schema}
+          uiSchema={uiSchema}
+          idSchema={idSchema}
+          formContext={formContext}
+          onChange={f => f}
+          requiredSchema={requiredSchema}/>
+    );
+
+    expect(tree.everySubTree('label')[0].text()).to.contain('Upload');
```

Comment:
The upload button is the label here, because we want to hide the usual file input element and just have a button that triggers the file explorer. It's weird.